applications/StructuralMechanicsApplication/custom_elements/membrane_element_3D.cpp

```diff
@@ -0,0 +1,1470 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Long Chen
+//
+
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "custom_elements/membrane_element_3D.hpp"
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+// Constructor
+MembraneElement3D::MembraneElement3D( IndexType NewId, GeometryType::Pointer pGeometry )
+    : Element( NewId, pGeometry )
+{
+}
+
+// Constructor
+MembraneElement3D::MembraneElement3D(IndexType NewId, GeometryType::Pointer pGeometry, PropertiesType::Pointer pProperties)
+    : Element( NewId, pGeometry, pProperties )
+{
+}
+
+//***********************************************************************************
+//***********************************************************************************
+
+Element::Pointer MembraneElement3D::Create(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties) const
+
+{
+    return boost::make_shared< MembraneElement3D >(NewId, GetGeometry().Create(ThisNodes), pProperties);
+}
+
+//***********************************************************************************
+//***********************************************************************************
+// Destructor
+MembraneElement3D::~MembraneElement3D()
+{
+}
+
+//***********************************************************************************
+//***********************************************************************************
+
+void MembraneElement3D::EquationIdVector(
+    EquationIdVectorType& rResult,
+    ProcessInfo& rCurrentProcessInfo)
+
+{
+  KRATOS_TRY;
+
+  /*
+  unsigned int number_of_nodes = GetGeometry().size();
+  unsigned int dim = number_of_nodes * 3;
+
+  if (rResult.size() != dim)
+      rResult.resize(dim);
+
+  for (unsigned int i = 0; i < number_of_nodes; ++i)
+  {
+      int index = i * 3;
+      rResult[index] = GetGeometry()[i].GetDof(DISPLACEMENT_X).EquationId();
+      rResult[index + 1] = GetGeometry()[i].GetDof(DISPLACEMENT_Y).EquationId();
+      rResult[index + 2] = GetGeometry()[i].GetDof(DISPLACEMENT_Z).EquationId();
+  }
+
+  Vector equationIdVector;
+  equationIdVector.resize(rResult.size());
+  for (unsigned int i = 0; i < rResult.size(); ++i)
+      equationIdVector[i] = rResult[i];
+  //KRATOS_WATCH(equationIdVector);
+  */
+
+  unsigned int NumNodes, LocalSize;
+  unsigned int LocalIndex = 0;
+
+  NumNodes = GetGeometry().size();
```

Comment:
You can use a template for the number of nodes of your membrane, that way you can use bounded_matrix that are faster